pyracf/common/security_admin.py

```diff
@@ -68,6 +75,8 @@ def __init__(
             self.__add_field_data(add_field_data)
         if overwrite_field_data is not None:
             self.__overwrite_field_data(overwrite_field_data)
+        if additional_secret_traits is not None:
+            self.__add_additional_secret_traits(additional_secret_traits)
```

Comment:
I think to reduce the likelihood of user error it makes sens for `additional_secret_traits` to just be a list. Then the user only needs to worry about the pyRACF trait name.

So instead of defining `additional_secret_traits` like this:

```python
user_admin = UserAdmin(additional_secret_traits={"omvs:uid": "uid"})
```

they can instead by added like this:

```python
user_admin = UserAdmin(additional_secret_traits=["omvs:uid"])
```

Then `__add_additional_secret_traits` would just need to lookup the IRRSMO00 key in `_valid_segment_traits`.

In short, do this to make the user's life easier.

Reply:
Agree, this makes more sense. With some tweaks, this brings the `__add_additional_secret_traits` function to the following.

```python
    def __add_additional_secret_traits(self, new_secrets: list):
        """Add additional fields to be redacted in logger output"""
        for secret in new_secrets:
            if secret in self._secret_traits:
                continue
            if ':' not in secret:
                continue
            segment = secret.split(':')[0]
            if segment not in self._valid_segment_traits:
                continue
            if secret not in self._valid_segment_traits[segment]:
                continue
            self._secret_traits[secret] = self._valid_segment_traits[segment][secret]
```

Reply:
Skipping bad input is OK for now, but as we discussed, we will definitely want to raise an exception or something in a future update when bad input is given.